### PR TITLE
Convert hundreds of tests to the new style

### PR DESCRIPTION
**Pull request by camsaul**
Remove the long-deprecated `expect-with-driver` and `expect-with-drivers` macros once and for all. Convert a couple dozen test namespaces to the new style.

This PR doesn't actually change any tests, just refactors things so they're written in the "new style" rather than `expect`-style

### CODE REVIEW

File: test/expectations.clj [hunk 1]
```diff
@@ -136,9 +136,8 @@
     (when-not (env/env :drivers)
       (t/testing "Don't write any new tests using expect!"
         (let [ee? (u/ignore-exceptions (require 'metabase-enterprise.core) true)]
-          ;; TODO - update the numbers for EE
-          (t/is (<= total-expect-forms (if ee? 871 842)))
-          (t/is (<= total-namespaces-using-expect (if ee? 84 80))))))))
+          (t/is (<= total-expect-forms (if ee? 549 524)))
+          (t/is (<= total-namespaces-using-expect (if ee? 59 56))))))))
```
Comment (camsaul) [inline test/expectations.clj:140]: dreams come true sometimes

File: test/metabase/test/data/datasets.clj [hunk 2]
```diff
@@ -63,54 +63,3 @@
...
-  each test.
-
-  DEPRECATED: Use `test-all-drivers` instead.
-
-    (deftest my-test
-      (datasets/test-all-drivers
-        (is (= ...))))"
-  {:style/indent 0}
-  [expected actual]
-  `(expect-with-drivers (tx.env/test-drivers) ~expected ~actual))
```
Comment (tsmacdonald) [inline test/metabase/test/data/datasets.clj:116]: :sunglasses:

**Review by tsmacdonald — APPROVED** [review]
I won't vouch for every token, but it looks good, and of course half the battle is having everything pass on CI—which it did.

**Review by robdaemon — APPROVED** [review]
I skimmed through this (it's a lot) but it looks good to me. Tests pass, coverage didn't change, I see nothing to complain about.

**Pull request merged**